doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+    from ray.rllib.utils.metrics import (
+        ENV_RUNNER_RESULTS,
+        EVALUATION_RESULTS,
+        EPISODE_RETURN_MEAN,
+    )
+    from ray import train, tune
 
-RLlib represents trajectory sequences (i.e., ``(s, a, r, s', ...)`` tuples) with `SampleBatch <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__ objects. Using a batch format enables efficient encoding and compression of experiences. During online training, RLlib uses `policy evaluation <rllib-concepts.html#policy-evaluation>`__ actors to generate batches of experiences in parallel using the current policy. RLlib also uses this same batch format for reading and writing experiences to offline storage.
+    # Configure the PPO algorithm.
+    config = (
+        PPOConfig()
+        .environment("CartPole-v1")
+        # Enable the new API stack.
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        .training(
+            lr=0.0003,
+            # Run 6 SGD minibatch iterations on a batch.
+            num_epochs=6,
+            # Weigh the value function loss smaller than 
+            # the policy loss.
+            vf_loss_coeff=0.01,
+        )
+        .rl_module(
+            model_config=DefaultModelConfig(
+                fcnet_hiddens=[32],
+                fcnet_activation="linear",
+                # Share encoder layers between value network
+                # and policy.
+                vf_share_layers=True,
+            ),
+        )
+    )
 
-Example: Training on previously saved experiences
--------------------------------------------------
+    # Define the metric to use for stopping.
+    metric = f"{EVALUATION_RESULTS}/{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}"
 
-.. note::
+    # Define the Tuner.
+    tuner = tune.Tuner(
+        "PPO",
+        param_space=config,
+        run_config=train.RunConfig(
+            stop={
+                metric: 450.0,
+            },
+            name="docs_rllib_offline_pretrain_ppo",
+            verbose=2,
+            checkpoint_config=train.CheckpointConfig(
+                checkpoint_frequency=1,
+                checkpoint_at_end=True,
+            ),
+        ),
+    )
+    analysis = tuner.fit()
 
-    For custom models and enviroments, you'll need to use the `Python API <rllib-training.html#basic-python-api>`__.
+    best_checkpoint = (
+        analysis
+        .get_best_result(
+            metric=metric,
+            mode="max"
+        )
+        .checkpoint.path
+    )
 
-In this example, we will save batches of experiences generated during online training to disk, and then leverage this saved data to train a policy offline using DQN. First, we run a simple policy gradient algorithm for 100k steps with ``"output": "/tmp/cartpole-out"`` to tell RLlib to write simulation outputs to the ``/tmp/cartpole-out`` directory.
+Example: Record expert data to local disk
```

Comment:
Maybe add a 1-2 sentences here between the paragraphs about that we are going to do next with the generated checkpoint.